mozci/data/sources/artifact/__init__.py

```diff
@@ -55,21 +57,24 @@ def _load_errorsummary(self, task_id) -> None:
                     group_results[group] = line["status"]
 
             elif line["action"] == "log":
-                if task_id not in self.TASK_ERRORS:
-                    self.TASK_ERRORS[task_id] = []
                 self.TASK_ERRORS[task_id].append(line["message"])
 
+        self.TASK_GROUPS[task_id] = {
+            group: result == "OK"
+            for group, result in group_results.items()
+            if result != "SKIP"
+        }
+
         missing_groups = groups - set(group_results)
         if len(missing_groups) > 0:
             logger.error(
                 f"Some groups in {task_id} are missing results: {missing_groups}"
             )
+            for group in missing_groups:
+                self.TASK_GROUPS[task_id][group] = False
```

Comment:
We shouldn't assume that a missing result means a failing test, in most cases if a group is missing it means it didn't run at all.

Reply:
Same reasoning as https://github.com/mozilla/mozci/pull/571#discussion_r760937698